### PR TITLE
add surepetcare doc

### PR DESCRIPTION
**Pull request by benleb**
**Description:** Component for smart pet/cat flaps from Sure Petcare

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#24426

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/surepetcare.markdown [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    description: The Sure Petcare Password
+    required: true
+    type: string
+  household_id:
+    description: The Sure Petcare household_id
+    required: true
+    type: integer
+  flaps:
+    description: The Sure Petcare flaps
+    required: true
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]:
```suggestion
    required: false
```

File: source/_components/surepetcare.markdown [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    type: map
+    keys:
+      id:
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]: add also `type: integer`

File: source/_components/surepetcare.markdown [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        description: The Sure Petcare id of a flap
+        required: true
+      name:
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]: Add also `type: string`

File: source/_components/surepetcare.markdown [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        description: A name for the flap
+        required: true
+  pets:
+    description: Pets managed by Sure Petcare flap(s)
+    required: true
+    type: map
+    keys:
+      id:
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]: Same

File: source/_components/surepetcare.markdown [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        description: The Sure Petcare id of a pet
+        required: true
+      name:
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]: Same

File: source/_components/surepetcare.markdown [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    keys:
+      id:
+        description: The Sure Petcare id of a flap
+        required: true
+      name:
+        description: A name for the flap
+        required: true
+  pets:
+    description: Pets managed by Sure Petcare flap(s)
+    required: true
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]:
```suggestion
    required: false
```

File: source/_components/surepetcare.markdown [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+# Example configuration.yaml entry
+surepetcare:
+  username: x@y.com
+  password: v3rys3cr3t!
+  household_id: 1337
+  flaps: [{id: 2337, name: Flap}]
+  pets: [{id: 3337, name: Pet}]
+```
+
+{% configuration %}
```
Comment (klaasnicolaas) [inline source/_components/surepetcare.markdown]: missing `device_class`, `icon` and `scan_interval`?

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
Because of age, I've rebased the PR.

**Review by frenck — APPROVED** [review]

**Pull request merged**